tests/foreman/cli/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,629 @@
+        self, module_architecture, module_azurerm_cr, module_os
+    ):
+        """ Cloud_init image along with username is being Create, Read, Update and
+        Delete in AzureRm compute resources
+
+        :id: 28e7ca34-f5f7-4712-9f1c-f4490ac2916a
+
+        :steps:
+            1. Create an AzureRm Compute Resource.
+            2. Create a Cloud_init based image in it.
+            3. List/info the created image
+            4. Update image name and username
+            5. Delete created image
+
+        :expectedresults: Cloud_init image should be created, list, updated and deleted in
+                          AzureRm CR along with username
+
+        :CaseImportance: Critical
+
+        :CaseLevel: Integration
+        """
+
+        # Create
+        img_name = gen_string('alpha')
+        username = gen_string('alpha')
+        img_ft = ComputeResource.image_create(
+            {
+                'name': img_name,
+                'operatingsystem-id': module_os.id,
+                'architecture-id': module_architecture.id,
+                'uuid': AZURERM_RHEL7_UD_IMG_URN,
+                'compute-resource': module_azurerm_cr.name,
+                'username': username,
+                'user-data': 'yes',
+            }
+        )[0]
+        assert img_ft['message'] == 'Image created.'
+        assert img_ft['name'] == img_name
+
+        # Info Image
+        img_info = ComputeResource.image_info(
+            {'name': img_name, 'compute-resource': module_azurerm_cr.name}
+        )[0]
+        assert img_info['operating-system'] == module_os.title
+        assert img_info['username'] == username
+        assert img_info['uuid'] == AZURERM_RHEL7_UD_IMG_URN
+        assert img_info['user-data'] == 'true'
+        assert img_info['architecture'] == module_architecture.name
+
+        # List image
+        list_img = ComputeResource.image_list({'compute-resource': module_azurerm_cr.name})
+        assert len(list_img) == 1
+        assert list_img[0]['name'] == img_name
+
+        # Update image
+        new_img_name = gen_string('alpha')
+        new_username = gen_string('alpha')
+        result = ComputeResource.image_update(
+            {
+                'name': img_name,
+                'compute-resource': module_azurerm_cr.name,
+                'new-name': new_img_name,
+                'username': new_username,
+            }
+        )[0]
+        assert result['message'] == 'Image updated.'
+        assert result['name'] == new_img_name
+
+        img_info = ComputeResource.image_info(
+            {'name': new_img_name, 'compute-resource': module_azurerm_cr.name}
+        )[0]
+        assert img_info['username'] == new_username
+
+        # Delete Image
+        result = ComputeResource.image_delete(
+            {'name': new_img_name, 'compute-resource': module_azurerm_cr.name}
+        )[0]
+        assert result['message'] == 'Image deleted.'
+        assert result['name'] == new_img_name
```

Comment:
Maybe I'm overlooking something but if the only difference between test_positive_crud_cloud_init_image and test_positive_crud_finish_template_image is in the constant passed to uuid (AZURERM_RHEL7_UD_IMG_URN vs AZURERM_RHEL7_FT_IMG_URN), then we could make it into one parametrized test for easier maintenance

Reply:
Good idea , let me give try in that way too.

Reply:
Done:
```
tests/foreman/cli/test_computeresource_azurerm.py::TestAzureRMComputeResourceTestCase::test_positive_image_crud[RedHat:RHEL:7-RAW:latest] PASSED                                       [ 50%]
tests/foreman/cli/test_computeresource_azurerm.py::TestAzureRMComputeResourceTestCase::test_positive_image_crud[RedHat:RHEL:7-RAW-CI:7.6.2019072418] PASSED                            [100%]

```